### PR TITLE
DEoptim sample

### PR DESCRIPTION
**Pull request by zfengms**

### CODE REVIEW

File: samples/README.md [hunk 1]
```diff
@@ -37,3 +37,7 @@ If you would like to see more samples, please reach out to [razurebatch@microsof
 7. **Using Azure Files** [(link)](./azure_files/readme.md)
 
    A quick introduction to setting up a distributed file system with Azure Files across all nodes in the cluster
+
+8. **DEoptim** [(link)](./deoptim/readme.md)
```
Comment (brnleehng) [inline samples/README.md]: **Using DEOptim** to follow the current pattern (verb "R package")

File: samples/deoptim/deoptim.r [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+library(devtools)
+install_github("azure/razurebatch")
+install_github("azure/doazureparallel")
+
+# import the doAzureParallel library and its dependencies
+library(doAzureParallel)
+
+credentialsFileName <- "credentials.json"
+clusterFileName <- "cluster.json"
```
Comment (brnleehng) [inline samples/deoptim/deoptim.r]: Do we need to provide a cluster configuration file?

File: samples/deoptim/deoptim.r [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  Sys.sleep(60)
+  1.0 + sum (100 * (x[-n]^2 - x[-1])^2 + (x[-1] - 1)^2)
+}
+
+# get some run-time on simple problems
+maxIt <- 5
+n <- 5
+
+options <- list(wait = TRUE, autoDeleteJob = TRUE)
+foreachArgs <- list(.options.azure = options, .packages = c('httr'))
```
Comment (brnleehng) [inline samples/deoptim/deoptim.r]: This sample requires httr?

File: samples/deoptim/deoptim.r [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+# === Caling DEoptim with doAzureParallel cluster as backend ===
+# ==============================================================
+
+## The below examples shows how the call to DEoptim can be parallelized
+## with doAzureParallel cluster as backend
+Genrose <- function(x) {
+  ## One generalization of the Rosenbrock banana valley function (n parameters)
+  n <- length(x)
+  ## simulate a long running task ...
+  Sys.sleep(60)
```
Comment (brnleehng) [inline samples/deoptim/deoptim.r]: Are there any other samples for running a long task?

Comment (brnleehng) [reply samples/deoptim/deoptim.r]: This sample looks promising? https://mintegration.eu/category/deoptim/? We should try running some benchmarks on this sample

File: samples/deoptim/readme.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# DEoptim sample
+
+This samples shows how to setup doAzureParallel as DEoptim's backend.
```
Comment (paselem) [inline samples/deoptim/readme.md]: "setup" -> "set up"

File: samples/deoptim/deoptim.r [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+  1.0 + sum (100 * (x[-n]^2 - x[-1])^2 + (x[-1] - 1)^2)
+}
+
+# get some run-time on simple problems
```
Comment (paselem) [inline samples/deoptim/deoptim.r]: This sentence doesn't make sense. Also, I'm not sure where we're getting the run time?

**Comment by brnleehng** [comment]
Stale

**Pull request closed without merging**